src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,498 @@
+ asynchronous function `getNewUsername()` and returns its result.
+  * Example return value from `changeUsername()`: `"jane_smith_92"` 
+  * Because `getNewUsername()` can encounter errors, the `changeUsername()` 
+  function must __catch and return any errors__ and it must __stringify the error before returning it__.
+* You can use the [toString()]({{site.dart_api}}/stable/dart-core/ArgumentError/toString.html) method to stringify both [Exceptions]({{site.dart_api}}/stable/dart-core/Exception-class.html) and [Errors.]({{site.dart_api}}/stable/dart-core/Error-class.html)
+
+<!-- [//]: https://gist.github.com/858f71f0ad0e70051999bcafa41806a3 -->
+
+<iframe 
+src="https://dartpad.dartlang.org/experimental/embed-new-dart.html?id=858f71f0ad0e70051999bcafa41806a3&theme=dark"
+frameborder="no"
+height="525"
+width="100%" >
+</iframe>
+
+### Exercise: Putting it all together
+
+It's time to practice what you've learned in one final exercise. 
+To simulate asynchronous operations, this exercise provides the asynchronous 
+functions `getUsername()` and `logoutUser()`. You don’t need to implement these
+provided functions. You don't need to implement `main()`.
+
+To simulate asynchronous operations, your code will call the following 
+functions, which are provided for you:
+
+|------------------+-----------------------------------+-------------|
+| Function         | Type signature                    | Description |
+|------------------|-----------------------------------|-------------|
+| getUsername()    | `Future<String> getUsername()`    |      Returns the name associated with the current user.|
+| logoutUser()    | `Future<String> logoutUser()`    |      Performs logout of current user and returns the username that was logged out.|
+{:.table .table-striped}
+
+Write the following:
+
+####  Part 1: `addHello()` 
+* Write a function `addHello()` that takes a single String argument 
+<!-- try to avoid should -->
+* `addHello()` returns its String argument surrounded by the text ‘Hello \<string\>'
+
+####  Part 2: `greetUser()` 
+* Write a function `greetUser()` that takes no arguments.
+* To obtain the username, `greetUser()` calls the provided asynchronous 
+function `getUsername()`
+* `greetUser()` creates a greeting for the user by calling `addHello()`, 
+passing it the username, and returning the result. 
+  * For example, if the username is "Jenny", `greetUser()` should create and
+   return the greeting "Hello Jenny"
+
+####  Part 3: `sayGoodbye()` 
+<!-- pull out descriptions of  -->
+* Write a function `sayGoodbye()` that does the following:
+* `sayGoodbye()` takes no arguments.
+* `sayGoodbye()` catches any errors.
+* `sayGoodbye()` calls the provided asynchronous function `logoutUser()`.
+<!-- italicize <result> -->
+* If `logoutUser()` succeeds, `sayGoodbye()` returns the string "\<result\> 
+Thanks, see you next time" where \<result\> is the String value returned by calling `logoutUser()`.
```

Comment:
returned by calling -> produced (eventually) by `logoutUser()`.

(or something else that acknowledges that logoutUser doesn't return a string value)